wolfssl/wolfcrypt/wc_port.h

```diff
@@ -278,37 +278,44 @@
 
 /* Reference counting. */
 typedef struct wolfSSL_Ref {
-/* TODO: use atomic operations instead of mutex. */
-#ifndef SINGLE_THREADED
+#if !defined(SINGLE_THREADED) && !defined(HAVE_C___ATOMIC)
     wolfSSL_Mutex mutex;
 #endif
     int count;
 } wolfSSL_Ref;
 
 #ifdef SINGLE_THREADED
-#define wolfSSL_RefInit(ref, err)           \
-    do {                                    \
-        (ref)->count = 1;                   \
-        *(err) = 0;                         \
-    }                                       \
-    while (0)
-
+#define wolfSSL_RefInit(ref, err) ({        \
+    (ref)->count = 1;                       \
+    *(err) = 0;                             \
+})
 #define wolfSSL_RefFree(ref)
-
-#define wolfSSL_RefInc(ref, err)            \
-    do {                                    \
-        (ref)->count++;                     \
-        *(err) = 0;                         \
-    }                                       \
-    while (0)
-
-#define wolfSSL_RefDec(ref, isZero, err)    \
-    do {                                    \
-        (ref)->count--;                     \
-        *(isZero) = ((ref)->count == 0);    \
-        *(err) = 0;                         \
-    }                                       \
-    while (0)
+#define wolfSSL_RefInc(ref, err)  ({        \
```

Comment:
This part (and wc_port.c) doesn't seem related to the sniffer changes? Or maybe I'm missing something? The code is newer than what we have.

Reply:
Correct, not part of sniffer. I added detection for atomic and fixed this TODO. I will separate the commits.

Reply:
Done